subsys/nfc/lib/platform_internal_thread.c

```diff
@@ -30,13 +30,30 @@ struct nfc_item_header {
 };
 
 #ifdef CONFIG_NFC_OWN_THREAD
-static K_SEM_DEFINE(cb_sem, 0, 1);
+/* Use counting semaphore, NFC events interrupt can go faster than the processing thread.
+ * All events must be processed.
+ */
+static K_SEM_DEFINE(cb_sem, 0, K_SEM_MAX_LIMIT);
 #endif /* CONFIG_NFC_OWN_THREAD */
 
 static bool buf_full;
 static nfc_lib_cb_resolve_t nfc_cb_resolve;
 RING_BUF_DECLARE(nfc_cb_ring, CONFIG_NFC_RING_SIZE);
 
+static void work_resubmit(struct k_work *work, struct ring_buf *buf)
+{
+	if (!IS_ENABLED(CONFIG_NFC_OWN_THREAD)) {
+		if (!ring_buf_is_empty(buf)) {
+			int ret = k_work_submit(work);
+			/* Work here can be only already scheduled from IRQ or it is running and has
```

Comment:
Let's change it to: In this case, work can be scheduled either from IRQ or to the queue running work items during the run.

Reply:
Ok, changed